proposals/269/transport.md

```diff
@@ -0,0 +1,74 @@
+---
+Status: IN PROGRESS  
+Author: Julien Le Dem (original issue) / Maciej Obuchowski (this document)  
+Created: 2021-10-18  
+Issue: [269](https://github.com/OpenLineage/OpenLineage/issues/269)
+---
+
+##Purpose
+
+The goal is to facilitate sending `OpenLineage` events over various protocols. 
+(HTTP, queues, ...) independently of the logic via exposing `Transport` interface.
+
+## Proposed interface
+
+In pseudocode:
+```
+interface Transport {
+  send(LineageEvent event);
+}
```

Comment:
Thinking of the ProxyBackend, it would be nice if this interface didn't require the service to deserialize the event just so it could be serialized again. I'm tempted to say this should just accept a String argument and let the callers be responsible for handling the JSON serialization. However, this would prohibit Transports from using some more efficient serialization, (e.g., Kafka supports Avro streams). I suppose if scale becomes an issue in the future, we can add a new method to allow more efficient passthrough.

Reply:
I agree there would a (de)serialization cost, but it be beneficial to quantify. Having a `RunEvent` makes it clear the metadata emitted along with the event for a given run. The other advantage is that the underlying method _controls_ the serialization, while also avoiding callers passing a string that may or may not be invalid.

As mentioned, methods can be added by modifying the interface of the `Transport` to support other serializations.